src/librustc_driver/driver.rs

```diff
@@ -414,6 +414,63 @@ impl<'a> CompileController<'a> {
     }
 }
 
+impl<'a> ::CompilerCalls<'a> for CompileController<'a> {
```

Comment:
Could you document this please? Why is this impl significant how should clients who are making a new controller do things?